NewForce-Cohort-9/tabloid-full-stack-api-all-stars
# Ticket #12 Edit a Category

### Description

The functionality to edit a category has been added. 

Given an admin has logged in to the app, when they are on the Category list page, they should be able to select an `Edit` button. When selected, the user will be prompted to update the Category information and hit Save to submit the changes. Given the user has decided not to edit the Category, they should should be able to Cancel the action and be redirected to the list page. 

[Trello Ticket #10](https://trello.com/c/u6UcRJXY/12-12-edit-a-category)

### Type of Change


### Testing Instructions

1. `git add` and `git commit` whatever you're working on on your branch
2. `git checkout main`
3. `git fetch --a`
4. `git checkout <il-category4>` 
5. If you don't see the most recent version of my files, `git pull origin <il-category4>`
6. If you haven't created the Tabloid database, please do so.
7. Use this to fill out tables and columns: [SQL data](https://github.com/NewForce-Cohort-9/tabloid-full-stack-api-all-stars/blob/main/SQL/01_Tabloid_Create_DB.sql)
8. Use this to seed the database: [SQL Seed Data](https://github.com/NewForce-Cohort-9/tabloid-full-stack-api-all-stars/blob/main/SQL/01_Tabloid_Create_DB.sql)
9. Click the execute with debugger solid green button in purple Visual Studio to launch the api
10. In your terminal, launch the program within the root folder with the command `npm start`
11. In browser, please go to `Login` on the nav bar and log in with this email: `foo@bar.com`
12. This should redirect you back to the home page. 
13. You should  see `Category Management` on the nav bar.
14. Select `Category Management` to be directed to `/category` where you should see a list of all categories ordered alphabetically.
15. Select the `Create Category` button and enter `Test` in the category name input and click `Save`
17. You should be directed back to the category list and see `Test` in the list.
18. Select the `edit` button next to the `Test` category.
19. You should be directed to an edit form where you can change the input to `TEST` and hit  `Save`
20. You should see the changes reflected in the category list.
21. Select the `edit` button next to the `Test` category.
22. Select `Nevermind` and be directed back to the category list and see that the `TEST` category is still part of the list.
23. Select the `delete` button next to the `Test` category.
25. You should be redirected to the category list and see that the `Test` category no longer exists.


### Checklist:
